tests/integration/test_integration.py

```diff
@@ -17,325 +18,331 @@
-            if passed:
-                break
-        if not passed:
-            raise Exception(
-                'HTML does not match\nActual:\n{}\n\nExpected:\n{}'.format(
-                    actual_cleaned,
-                    expected_cleaned
-                )
+def test_wildcard_callback(dash_duo):
+    app = Dash(__name__)
+    app.layout = html.Div([
+        dcc.Input(id='input', value='initial value'),
+        html.Div(
+            html.Div([
+                1.5,
+                None,
+                'string',
+                html.Div(id='output-1', **{'data-cb': 'initial value',
+                                           'aria-cb': 'initial value'})
+            ])
+        )
+    ])
+
+    input_call_count = Value('i', 0)
+
+    @app.callback(Output('output-1', 'data-cb'), [Input('input', 'value')])
+    def update_data(value):
+        input_call_count.value += 1
+        return value
+
+    @app.callback(Output('output-1', 'children'),
+                  [Input('output-1', 'data-cb')])
+    def update_text(data):
+        return data
+
+    dash_duo.start_server(app)
+    dash_duo.wait_for_text_to_equal('#output-1', 'initial value')
+    dash_duo.percy_snapshot(name='wildcard-callback-1')
+
+    input1 = dash_duo.find_element('#input')
+    dash_duo.clear_input(input1)
+    input1.send_keys('hello world')
+
+    dash_duo.wait_for_text_to_equal('#output-1', 'hello world')
+    dash_duo.percy_snapshot(name='wildcard-callback-2')
+
+    # an initial call, one for clearing the input
+    # and one for each hello world character
+    assert input_call_count.value == 2 + len('hello world')
+
+    assert not dash_duo.get_logs()
+
+
+def test_aborted_callback(dash_duo):
+    """
+    Raising PreventUpdate OR returning no_update
+    prevents update and triggering dependencies
+    """
+
+    initial_input = 'initial input'
+    initial_output = 'initial output'
+
+    app = Dash(__name__)
+    app.layout = html.Div([
+        dcc.Input(id='input', value=initial_input),
+        html.Div(initial_output, id='output1'),
+        html.Div(initial_output, id='output2'),
+    ])
+
+    callback1_count = Value('i', 0)
+    callback2_count = Value('i', 0)
+
+    @app.callback(Output('output1', 'children'), [Input('input', 'value')])
+    def callback1(value):
+        callback1_count.value += 1
+        if callback1_count.value > 2:
+            return no_update
+        raise PreventUpdate("testing callback does not update")
+        return value
+
+    @app.callback(
+        Output('output2', 'children'), [Input('output1', 'children')]
+    )
+    def callback2(value):
+        callback2_count.value += 1
+        return value
+
+    dash_duo.start_server(app)
+
+    input_ = dash_duo.find_element('#input')
+    input_.send_keys('xyz')
+    dash_duo.wait_for_text_to_equal('#input', 'initial inputxyz')
+
+    # callback1 runs 4x (initial page load and 3x through send_keys)
```

Comment:
I would put the comment as an assert msg, so you can see that directly in the assertion exception.   the comment does show up too with the code snippet.